leapsecs_read.c

```diff
@@ -0,0 +1,50 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+extern int errno;
+#include "tai.h"
+#include "leapsecs.h"
+
+struct tai *leapsecs = 0;
+int leapsecs_num = 0;
+
+int leapsecs_read()
+{
+  int fd;
+  struct stat st;
+  struct tai *t;
+  int n;
+  int i;
+  struct tai u;
+
+  fd = open("/etc/leapsecs.dat",O_RDONLY | O_NDELAY);
```

Comment:
I usually do this before:

```c
#ifndef O_CLOEXEC
#define O_CLOEXEC 0
#endif
```

And then pass O_CLOEXEC to all these things just out of paranoia.

Reply:
The libtai-0.60 library was designed to be small. It intentionally avoids a lot of the syntactic system libraries that qmail relies upon: open.h, substdio.h/buffer.h, and so on. Perhaps O_CLOEXEC doesn't belong here, but in open_read.c, which would then replace the open() here.

Reply:
One needs the file descriptor in a child process e.g. when passing the mail body to qmail-remote, so always using O_CLOEXEC is probably wrong.